applications/flash_attention_v2/collective/xe_flash_attn_decode_mma.hpp

```diff
@@ -238,9 +241,11 @@ struct FlashDecodeMma<gemm::MainloopIntelXeXMX16<Stages>, ProblemShapeType_, Ele
     Tensor tCgK = thread_mma_k.partition_B(gK);
 
     // Create fragments
-    Tensor tCrQ = make_tensor<ElementQ>(make_fragment_layout(params.gmem_tiled_copy_q, take<0,3>(tCgQ.shape())));
-    Tensor tCrK = make_tensor<ElementK>(make_fragment_layout(gmem_tiled_copy_k, take<0,3>(tCgK.shape())));
-    
+    using TCrQ_Type = cute::conditional_t<is_fp8_v<ElementQ>, uint8_t, ElementQ>;
```

Comment:
can we avoid that by telling the copy to treat fp8 as unit8_t? Then we wouldn't need to pass ElementQ/K to convert_FP8_to_FP16 because the tensor had already the correct type.

Reply:
All configuration currently have this. including gemm. @sanchitintel  can answer better why (I think it is mainly performance related), but this can be handled in a separate PRs for all example using it and then removing the static assert requirement for uint8_t inside fp8tofp16

Reply:
Sorry, I just came across this notification.

> can we avoid that by telling the copy to treat fp8 as unit8_t?

@rolandschulz, is your recommendation geared towards simplifying code?
If yes, please note that if `operator half` would be used instead of `cutlass::NumericArrayConverter` for FP8_E5M2 to FP16 conversion, then the conversion would be slower.

Thanks!